pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -337,11 +337,10 @@ func CreateCluster(l *logger.Logger, opts []*ClusterCreateOpts) error {
 		return vm.Name(name, nodesCreated)
 	}
 	for _, o := range opts {
-		providerCount := len(o.CreateOpts.VMProviders)
-		if providerCount == 0 {
-			return errors.New("no VMProviders configured")
+		if err := validateProviders(o.CreateOpts.VMProviders); err != nil {
```

Comment:
Why couldn't we do this level of validation immediately after parsing CLI args, i.e., in `validateAndConfigure`?

Reply:
In all honesty I was unaware of `validateAndConfigure`, but I don't have any strong opinions against moving this there; considering this issue is more prone to CLI user input, I'll move the validation to `validateAndConfigure`.

Reply:
It just makes sense to validate user input prior to taking any action(s). I am not surprised you were unaware of `validateAndConfigure` since it was added fairly recently. Historically, `roachprod` has lacked input validation.

Reply:
Updated, and moved.